tests/factories.py

```diff
@@ -283,3 +283,40 @@ class Params:
     level = factory.Sequence(lambda n: n)
     path = factory.Sequence(lambda n: (f"path-{n}",))
     navlink = factory.SubFactory(NavlinkFactory)
+
+
+# The attributes of these classes are generators for the attributes of the meta class
+class IndexParsedListItemFactory(factory.Factory):
+    # Docstrings have been abbreviated for factories, checking for docstrings on model attributes
+    # can be skipped.
+    """Generate index._ParsedListItems."""  # noqa: DCO060
+
+    class Meta:
+        """Configuration for factory."""  # noqa: DCO060
+
+        # Need access to be able to generate instances
+        model = index._ParsedListItem  # pylint: disable=protected-access
+        abstract = False
+
+    whitespace_count = factory.Sequence(lambda n: n)
+    reference_title = factory.Sequence(lambda n: f"reference title {n}")
+    reference_value = factory.Sequence(lambda n: f"reference value {n}")
+    rank = factory.Sequence(lambda n: n)
```

Comment:
I don't think that is how factoryboy works, I'm guessing that what happens is that the sequence is called whenever the factory method is executed so perhaps the `whitespace_count` would probably work but the `reference_title` would have a `str` representation of the sequence rather than the number. These are class attributes, not instance attributes